### PR TITLE
feat: add exclude arg and use to ignore

### PR DESCRIPTION
**Pull request by gitphill**
- [X] Ready for review
- [X] Follows [CONTRIBUTING](https://github.com/snyk/snyk/blob/master/.github/CONTRIBUTING.md) rules
- [ ] Reviewed by Snyk internal team

#### What does this PR do?
Ignore directories when using --all-projects.
Added tests for test and monitor with --exclude.

#### How should this be manually tested?
`snyk test --all-projects --detection-depth=2 --exclude=example-dir,example-2`

#### What are the relevant tickets?
BST-1126

### CODE REVIEW

File: src/lib/plugins/get-deps-from-plugin.ts [hunk 1]
```diff
@@ -25,7 +25,13 @@ export async function getDepsFromPlugin(
 
   if (options.allProjects) {
     const levelsDeep = options.detectionDepth || 1; // default to 1 level deep
-    const targetFiles = await find(root, [], AUTO_DETECTABLE_FILES, levelsDeep);
+    const ignore = options.exclude ? options.exclude.split(',') : [];
```
Comment (lili2311) [inline src/lib/plugins/get-deps-from-plugin.ts]: what is `options.exclude` is badly formatted?

Comment (gitphill) [reply src/lib/plugins/get-deps-from-plugin.ts]: lets add a test :)

Comment (lili2311) [reply src/lib/plugins/get-deps-from-plugin.ts]: `--exclude folder1,folder2` or even ` --exclude=a/b/c` as long as it doesn't break the current test run.

Comment (gitphill) [reply src/lib/plugins/get-deps-from-plugin.ts]: ok added some validation and tests

File: src/lib/errors/exclude-flag-invalid-input.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import { CustomError } from './custom-error';
+
+export class ExcludeFlagInvalidInputError extends CustomError {
```
Comment (lili2311) [inline src/lib/errors/exclude-flag-invalid-input.ts]: I think we can merge these two errors and provide the custom string into the constructor at call site?

File: src/lib/errors/option-missing-error.ts [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import { CustomError } from './custom-error';
+
+export class OptionMissingErrorError extends CustomError {
```
Comment (lili2311) [inline src/lib/errors/option-missing-error.ts]: ❤️

File: test/acceptance/cli-monitor/cli-monitor.all-projects.spec.ts [hunk 4]
```diff
@@ -80,7 +80,6 @@ export const AllProjectsTests: AcceptanceTests = {
         'calls maven plugin twice',
       );
       // maven
-      console.log(result);
```
Comment (lili2311) [inline test/acceptance/cli-monitor/cli-monitor.all-projects.spec.ts]: 😅

File: src/lib/errors/exclude-flag-invalid-input.ts (hunk 2 above)
Comment (lili2311) [reply src/lib/errors/exclude-flag-invalid-input.ts]: can do it later

**Pull request merged**